src/osdc/Objecter.cc

```diff
@@ -1006,7 +1006,7 @@ void Objecter::handle_osd_map(MOSDMap *m)
   }
 
   bool was_pauserd = osdmap->test_flag(CEPH_OSDMAP_PAUSERD);
-  bool was_full = _osdmap_full_flag();
+  bool was_full = _osdmap_full_flag() || _osdmap_pool_full_flag();
```

Comment:
the logic here will need a bit of adjustment - we should block writes for pools that were full, not all pools when an unrelated pool is full